chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -572,9 +594,15 @@ public void joinChannel(String channelName) {
      */
     public void leaveChannel(String channelName) {
         String lowerChannelName = channelName.toLowerCase();
-        if (channelCache.remove(lowerChannelName)) {
+        if (currentChannels.remove(lowerChannelName)) {
             sendCommand("part", "#" + lowerChannelName);
             log.debug("Leaving Channel [{}].", lowerChannelName);
+
+            // clear cache
+            if (channelNameToChannelId.containsKey(lowerChannelName)) {
+                channelIdToChannelName.remove(channelNameToChannelId.get(lowerChannelName));
+                channelNameToChannelId.remove(lowerChannelName);
+            }
```

Comment:
Could be replaced by:
```java
String cachedId = channelNameToChannelId.remove(lowerChannelName);
if (cachedId != null) channelIdToChannelName.remove(cachedId);
```
for a marginal performance gain and perhaps greater robustness to race conditions when multiple threads are doing joins/parts simultaneously